src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,88 @@
+package parser;
+
+import com.google.gson.JsonSyntaxException;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import shop.Cart;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Tag("JsonParserTest")
+@DisplayName("JsonParser Class Tests")
+class JsonParserTest {
+    Parser parser;
+
+    @BeforeEach
+    public void setup() {
+        parser = new JsonParser();
+    }
+
+    @DisplayName("Should Write To File")
+    @Test
+    public void shouldWriteToFile() {
+        Cart cart = new Cart("test-cart");
+        parser.writeToFile(cart);
+        // have to use main resources folder as writeToFile method creates files there, not in test resources
+        File file = new File("src/main/resources/" + cart.getCartName() + ".json");
+        // checking that file is not empty
+        assertNotEquals(0, file.length());
+        // delete file on exit, otherwise test will pass regardless in future test executions
+        file.deleteOnExit();
```

Comment:
the file will remain undeleted if the assertion fails. It is better to move it to the `@After...` methods